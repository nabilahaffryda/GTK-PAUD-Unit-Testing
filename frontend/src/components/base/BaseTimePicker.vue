<template>
  <v-row dense no-gutters>
    <v-col cols="12" md="4" sm="6">
      <v-select
        v-model="date.hour"
        placeholder="jam"
        :outlined="outlined"
        :disabled="disabled"
        :dense="dense"
        :items="hours"
        required
      />
    </v-col>
    <v-col class="mx-md-1" cols="12" md="4" sm="6">
      <v-select
        v-model="date.minute"
        placeholder="menit"
        :outlined="outlined"
        :disabled="disabled"
        :dense="dense"
        :items="minutes"
        required
      />
    </v-col>
    <v-col class="mt-3 mx-2" cols="12" md="2" sm="2"> WIB </v-col>
  </v-row>
  <!--  <v-dialog ref="dialog" v-model="modal" :return-value.sync="date" persistent width="290px">-->
  <!--    <template v-slot:activator="{ on }">-->
  <!--      <v-text-field-->
  <!--        :label="label"-->
  <!--        :prepend-icon="useicon ? 'mdi-calendar' : ''"-->
  <!--        readonly-->
  <!--        :value="date"-->
  <!--        v-on="on"-->
  <!--        :disabled="disabled"-->
  <!--        :dense="dense"-->
  <!--        :single-line="singleLine"-->
  <!--        :outlined="outlined"-->
  <!--        :error-messages="errors"-->
  <!--      />-->
  <!--    </template>-->
  <!--    <v-time-picker scrollable v-model="date" format="24hr">-->
  <!--      <v-spacer />-->
  <!--      <v-btn text color="primary" @click="modal = false">Cancel</v-btn>-->
  <!--      <v-btn text color="primary" @click="$refs.dialog.save(date)">OK</v-btn>-->
  <!--    </v-time-picker>-->
  <!--  </v-dialog>-->
</template>
<script>
export default {
  name: 'TimePicker',
  props: ['value', 'label', 'min', 'max', 'errors', 'disabled', 'useicon', 'dense', 'singleLine', 'outlined'],
  data() {
    return {
      modal: false,
      date: {
        hour: this.value ? this.value.split(':')[0] : '',
        minute: this.value ? this.value.split(':')[1] : '',
      },
    };
  },
  computed: {
    formatDate() {
      return this.date.hour || this.date.minute ? `${this.date.hour}:${this.date.minute || '00'}` : null;
    },

    hours() {
      let temp = [];
      for (let i = 1; i <= 23; i++) {
        if (i < 10) temp.push(`0${i}`);
        else temp.push(`${i}`);
      }
      return temp;
    },

    minutes() {
      let temp = [];
      for (let i = 0; i <= 55; i++) {
        if (i === 0 || i % 5 === 0) {
          if (i < 10) temp.push(`0${i}`);
          else temp.push(`${i}`);
        }
      }

      return temp;
    },
  },
  methods: {
    mutate() {
      this.$emit('input', this.formatDate);
    },
    setForm(value) {
      this.$set(this.date, 'hour', value ? value.split(':')[0] : '');
      this.$set(this.date, 'minute', value ? value.split(':')[1] : '');
    },
  },
  watch: {
    date: {
      handler() {
        this.mutate();
      },
      deep: true,
    },
    value: 'setForm',
  },
};
</script>
