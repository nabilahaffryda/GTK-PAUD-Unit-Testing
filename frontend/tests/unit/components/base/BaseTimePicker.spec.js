import { mount, createLocalVue } from "@vue/test-utils";
import BaseTimePicker from "@/components/base/BaseTimePicker.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseTimePicker.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('render time picker', () => {
        const VSelect = {
            props: ['value', 'label', 'min', 'max', 'errors', 'disabled', 'useicon', 'dense', 'singleLine', 'outlined'],
            template: '<div><slot :dense="dense" /></div>'
        }

        const wrapper = mount(BaseTimePicker, {
            localVue,
            vuetify,
            stubs: {
                VSelect
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
