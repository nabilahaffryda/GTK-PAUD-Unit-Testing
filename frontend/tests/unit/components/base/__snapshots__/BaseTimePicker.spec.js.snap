// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTimePicker.vue render time picker 1`] = `
<div class="row no-gutters row--dense">
  <div class="col-sm-6 col-md-4 col-12">
    <div placeholder="jam" items="01,02,03,04,05,06,07,08,09,10,11,12,13,14,15,16,17,18,19,20,21,22,23" required="required"></div>
  </div>
  <div class="mx-md-1 col-sm-6 col-md-4 col-12">
    <div placeholder="menit" items="00,05,10,15,20,25,30,35,40,45,50,55" required="required"></div>
  </div>
  <div class="mt-3 mx-2 col-sm-2 col-md-2 col-12"> WIB </div>
</div>
`;
